root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs

```diff
@@ -99,126 +109,223 @@ private void InitializeComponent()
+            ThreadStart threadStart = new ThreadStart(() => StartByThreadPool(_asyncParameterValue));
+            _workerThread = new Thread(threadStart);
+            _workerThread.Start();
+
+            //Update in mainthread
+            _asyncParameterValue = new AsyncProcessingServiceParameterValue("AsyncProcessingService", "Update", "Update", "SQL",
+                                                                            new MyUserInfo("AsyncProcessingService", "AsyncProcessingService"));
+
+            DbEnum.IsolationLevelEnum iso = DbEnum.IsolationLevelEnum.DefaultTransaction;
+
+            asyncReturnValue = (AsyncProcessingServiceReturnValue)myBusiness.DoBusinessLogic((AsyncProcessingServiceParameterValue)_asyncParameterValue, iso);
         }
 
+        #endregion
+
+        #region StartByThreadPool
+
         /// <summary>
         /// Calls callback method using Threadpool worker item.
         /// </summary>
-        /// <param name="check"></param>
+        /// <param name="_asyncParameterValue"></param>
         /// <returns></returns>
-        public bool StartByThreadPool(AsyncProcessingServiceParameterValue.AsyncStatus check)
+        public bool StartByThreadPool(AsyncProcessingServiceParameterValue _asyncParameterValue)
+        {
+            ThreadPool.QueueUserWorkItem(new WaitCallback(InvokeController), (object)_asyncParameterValue.ProcessName);
+            return true;
+        }
+
+        #endregion
+
+        #region InvokeController
+
+        /// <summary>
+        /// Updates using LayerB through callcontroller class
+        /// </summary>
+        /// <param name="_asyncParameterValue"></param>
+        private void InvokeController(object _asyncParameterValue)
         {
-            int maxThreadCount;
+            AsyncProcessingServiceReturnValue asyncReturnValue;
+            object deserializeDatas = DeserializeFromBase64(_asyncData.Data);
+            string[] strDatas = deserializeDatas.ToString().Split('/');
 
-            lock (this)
+            _asyncData.UserId = strDatas.GetValue(0).ToString();
+            _asyncData.RegistrationDateTime = Convert.ToDateTime(strDatas.GetValue(2));
+            _asyncData.ExecutionStartDateTime = Convert.ToDateTime(strDatas.GetValue(3));
+            _asyncData.NumberOfRetries = Convert.ToInt32(strDatas.GetValue(4));
+            _asyncData.CompletionDateTime = Convert.ToDateTime(strDatas.GetValue(5));
+            string[] strReservedArea = strDatas.GetValue(6).ToString().Split('-');
+            _asyncData.ReservedArea = strReservedArea.GetValue(0) + " " + strReservedArea.GetValue(1);
+
+            // InProcess Call using CallController for updating service in database
+            CallController callController = new CallController(this._asyncParameterValue.User);
+            asyncReturnValue = (AsyncProcessingServiceReturnValue)callController.Invoke(_asyncParameterValue.ToString(), _asyncData);
```

Comment:
You need to specify the MyParameterValue that was got by deserializing AsyncProcessingServiceReturnValue.Data to the second argument.